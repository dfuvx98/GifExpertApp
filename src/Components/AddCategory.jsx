import { useState } from "react"


export const AddCategory = ({onNewCategory}) => {
  
    const [inputValue, setinputValue] = useState('One Punch')

    const onInputChange = ({target}) =>{
        setinputValue(target.value);
    };

    const onSubmit =(event) =>{
        event.preventDefault();
        if(inputValue.trim().length<=1)return;
        //setcategories( (categories => [inputValue, ...categories]));
        onNewCategory(inputValue.trim() );
        setinputValue('');
    }

    return (
    <form onSubmit={ onSubmit}>
        <input 
        type="text" 
        placeholder="Buscar gifs"
        value = {inputValue}
        onChange={onInputChange}
        />
    </form>
  )
}
