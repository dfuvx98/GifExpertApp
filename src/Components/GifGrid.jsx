import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

//import {getGifs} from '../helpers/getGifs';




export const GifGrid = ({category,remove}) =>{

    const { images, isLoading } = useFetchGifs ( category );
    
    /*
*/
    return (
        <>
            <h3>{ category }</h3>
            
            {
                isLoading && (<h2>Cargando...</h2>)
                
            }
            
            
            <div className="card-grid">
                {
                    images.map((image) =>(
                        <GifItem 
                        key={image.id}
                        {...image }
                        
                        />
                    ))
                }
            
            </div>
            <div>
                <button onClick={() => remove(category)}>Quitar</button>
            </div>
        </>
  )
}

/*const getGifs = async(category) =>{

    const url = `https://api.giphy.com/v1/gifs/search?api_key=UxpAKOsLTci9tZCEndNn97sIOk7j8r48&q=${ category }&limit=20` 
    const resp = await fetch(url);
    const { data =[]} = await resp.json();
    
    const gifs = data.map(img =>({
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url
    }))
    console.log(gifs);
    return gifs;
}


export const GifGrid = ({category}) => {
    
    
    getGifs(category);

*/  