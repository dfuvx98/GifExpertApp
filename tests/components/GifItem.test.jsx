import { render,screen } from "@testing-library/react";
import { GifItem } from "../../src/Components";

describe('Pruebas en <GifItem />', () =>{
        const title = 'Saitama';
        const url = 'https://one-punch.com/saitama.jpg';
        test('debe de hacer match cn el snapshot', () => {

            const { container } =  render( <GifItem title= {title} url= {url} />);
            expect( container ).toMatchSnapshot();
            
    });
    
    test('debe de mosrar la imagen con el URL y el ALT indicado', () =>{
         render( <GifItem title= {title} url= {url} />);
         //screen.debug();
         /*expect(screen.getByRole('img').src).toBe(url);
         expect(screen.getByRole('img').alt).toBe(title);*/
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

    });

    test('debe de mostrar el tÃ­tulo en el componente', () =>{

        render( <GifItem title={title} url={url} />);
        expect( screen.getByText(title)).toBeTruthy();
        //a
    })
});
 