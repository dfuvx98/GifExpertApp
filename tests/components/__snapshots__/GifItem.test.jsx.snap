// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <GifItem /> debe de hacer match cn el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
